src/main/java/net/minecraftforge/registries/RegistryObject.java

```diff
@@ -51,27 +56,75 @@
      *
-     * @param name The name of the object to look up in a forge registry
-     * @param key The key of the forge registry to grab from {@link RegistryManager#ACTIVE}
-     * @param modid The mod id calling context
-     * @return A {@link RegistryObject} that stores the value of an object in a registry once it is ready
+     * @param name the name of the object to look up in a registry
+     * @param registryKey the key of the registry. Supports lookups on {@link BuiltinRegistries}, {@link Registry}, and {@link RegistryManager#ACTIVE}.
+     * @param modid the mod id calling context
+     * @return a {@link RegistryObject} that stores the value of an object from a registry once it is ready
+     * @see #optional(ResourceLocation, ResourceKey, String)
+     * @see #of(ResourceLocation, IForgeRegistry)
+     * @see #of(ResourceLocation, ResourceLocation, String)
      */
-    public static <T, U extends T> RegistryObject<U> of(final ResourceLocation name, final ResourceKey<? extends Registry<T>> key, String modid) {
-        return new RegistryObject<>(name, key.location(), modid);
+    public static <T, U extends T> RegistryObject<U> of(final ResourceLocation name, final ResourceKey<? extends Registry<T>> registryKey, String modid) {
+        return new RegistryObject<>(name, registryKey.location(), modid, false);
     }
 
     /**
-     * Factory for a {@link RegistryObject} that stores the value of an object in a registry once it is ready.
+     * Factory for a {@link RegistryObject} that stores the value of an object from a registry once it is ready.
+     * <p>
+     * If a registry with the given key cannot be found, it will be silently ignored and this RegistryObject will not be filled.
+     * Use {@link #of(ResourceLocation, ResourceKey, String)} for RegistryObjects that should throw exceptions on missing registry.
      *
-     * @param name The name of the object to look up in a forge registry
-     * @param registryName The name of the forge registry to grab from {@link RegistryManager#ACTIVE}
-     * @param modid The mod id calling context
-     * @return A {@link RegistryObject} that stores the value of an object in a registry once it is ready
+     * @param name the name of the object to look up in a registry
+     * @param registryKey the key of the registry. Supports lookups on {@link BuiltinRegistries}, {@link Registry}, and {@link RegistryManager#ACTIVE}.
+     * @param modid the mod id calling context
+     * @return a {@link RegistryObject} that stores the value of an object from a registry once it is ready
+     * @see #of(ResourceLocation, ResourceKey, String)
+     * @see #of(ResourceLocation, IForgeRegistry)
+     * @see #of(ResourceLocation, ResourceLocation, String)
+     */
+    public static <T, U extends T> RegistryObject<U> optional(final ResourceLocation name, final ResourceKey<? extends Registry<T>> registryKey, String modid) {
```

Comment:
First, `ofOptional` would probably be more conforming to the current conventional names of the factories (akin to `Optional.of` and `Optional.ofNullable`).
Formatting: braces on newlines.
```suggestion
    public static <T, U extends T> RegistryObject<U> ofOptional(final ResourceLocation name, final ResourceKey<? extends Registry<T>> registryKey, String modid)
    {
```